examples/7_extended_write_serial.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+"""
+This file is part of the openPMD-api.
+
+Copyright 2018 openPMD contributors
+Authors: Axel Huebl, Fabian Koller
+License: LGPLv3+
+"""
+from openPMD import Series, Access_Type, Dataset, Mesh_Record_Component, \
+    Unit_Dimension, Datatype, determine_datatype
+import numpy as np
+
+
+SCALAR = Mesh_Record_Component.SCALAR
+
+
+if __name__ == "__main__":
+    # open file for writing
+    f = Series(
+        "working/directory/2D_simData_py.h5",
+        Access_Type.create
+    )
+
+    # all required openPMD attributes will be set to reasonable default values
+    # (all ones, all zeros, empty strings,...)
+    # manually setting them enforces the openPMD standard
+    f.set_meshes_path("custom_meshes_path")
+    f.set_particles_path("long_and_very_custom_particles_path")
+
+    # it is possible to add and remove attributes
+    f.set_comment("This is fine and actually encouraged by the standard")
+    f.set_attribute(
+        "custom_attribute_name",
+        "This attribute is manually added and can contain about any datatype "
+        "you would want"
+    )
+    # note that removing attributes required by the standard typically makes
+    # the file unusable for post-processing
+    f.delete_attribute("custom_attribute_name")
+
+    # setting attributes can be chained in JS-like syntax for compact code
+    tmpItObj = f.iterations[1] \
+        .set_time(42.0) \
+        .set_dt(1.0) \
+        .set_time_unit_SI(1.39e-16)
+    # everything that is accessed with [] should be interpreted as permanent
+    # storage the objects sunk into these locations are deep copies
+    del tmpItObj
+    f.iterations[2].set_comment("This iteration will not appear in any output")
+    del f.iterations[2]
+
+    # this is a reference to an iteration
+    reference = f.iterations[1]
+    reference.set_comment("Modifications to a reference will always be visible"
+                          " in the output")
+    del reference
+
+    # alternatively, a copy may be created and later re-assigned to
+    # f.iterations[1]
+    copy = f.iterations[1]  # TODO .copy()
```

Comment:
This is actually outdated in the C++ example. Please add the **TODO** there as well.

Reply:
Yes, I thought so. Let's update both.